Userland/Applications/Dictionary/Dictionary.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2021, Robin Allen <r@foon.uk>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/MappedFile.h>
+#include <AK/String.h>
+#include <AK/StringView.h>
+
+namespace Dictionary {
+
+class Dictionary {
+public:
+    Dictionary(char const* filename);
+
+    String definition_of(int word_index) const;
+
+    StringView word_at_index(int word_index) const;
+
+    int count_words() const { return m_num_words; }
+
+    void prefix_query(
+        String query,
+        int start_index,
+        int& first_index,
+        int& num_indices) const;
+
+private:
+    template<typename T>
+    struct Offset {
+        uint32_t value;
+    };
+
+    struct Definition {
+        char part_of_speech;
+        char text[];
+    } __attribute__((packed));
+
+    struct Sense {
+        Offset<Definition> definition_offset;
+    } __attribute__((packed));
+
+    struct WordData {
+        uint8_t num_senses;
+        Sense senses[];
+    } __attribute__((packed));
+
+    typedef char IndexEntry[16];
+    typedef Offset<WordData> WordDataIndexEntry;
+
+    struct Header {
+        uint32_t num_words;
+        uint32_t max_word_length;
+        Offset<IndexEntry> index_offset;
+        Offset<WordDataIndexEntry> word_data_index_offset;
+    };
+
+    RefPtr<MappedFile> m_dictionary_file;
+    uint8_t const* m_dictionary_bytes;
+
+    int m_num_words;
+    int m_max_word_length;
```

Comment:
primitive types (arithmetic types, pointers, etc) conventionally use designated initializers in the class declaration.

```c++
u8 const* m_dictionary_bytes { nullptr };

int m_num_words { 0 };
int m_max_word_length { 0 };
```

Reply:
Done.